### ISSUE TITLE
[apply retry] retry on etcdserver leader changed errors

### ISSUE THREAD

**Issue by cameronbraid** [vmware-tanzu/carvel-kapp#106]
This tends to happen when I deploy my full dev cluster form scratch which is a lot of resources

```
kapp: Error: waiting on reconcile ......: etcdserver: leader changed
```

It would be good if kapp detected this error and retried

**Comment by cppforlife** [comment]
make sense. (though im surprised that client-go does not already retry on that).

**Comment by cppforlife** [comment]
potentially similar error:

> 00:09:17.924 kapp: Error: waiting on reconcile deployment/prometheus-server (apps/v1) namespace: monitoring: errored: Getting resource deployment/prometheus-server (apps/v1) namespace: monitoring: etcdserver: request timed out (reason: )

**Comment by renuy** [comment]
available in release v0.40.0
